src/battle_script_commands.c

```diff
@@ -3495,6 +3503,10 @@ void SetMoveEffect(bool32 primary, u32 certain)
                 gBattleMons[gBattlerTarget].status2 |= STATUS2_ESCAPE_PREVENTION;
                 gBattleMons[gBattlerAttacker].status2 |= STATUS2_ESCAPE_PREVENTION;
                 break;
+            case MOVE_EFFECT_SCALE_SHOT:
```

Comment:
this should include the `if (!NoAliveMonsForEitherParty())` check for other stat-changing move effects, which prevent the stat changes from occurring if the battle would end. See V_CREATE for example